E0026 needs to be updated to new format

From: src/test/compile-fail/E0026.rs

Error E0026 needs a span_label, updating it from:

```
error[E0026]: struct `Thing` does not have a field named `z`
  --> src/test/compile-fail/E0026.rs:19:23
   |
19 |         Thing { x, y, z } => {} //~ ERROR E0026
   |                       ^
```

To:

```
error[E0026]: struct `Thing` does not have a field named `z`
  --> src/test/compile-fail/E0026.rs:19:23
   |
19 |         Thing { x, y, z } => {} //~ ERROR E0026
   |                       ^ struct `Thing` does not have field `z`
```

--- a/src/librustc_typeck/check/_match.rs
+++ b/src/librustc_typeck/check/_match.rs
@@ -682,10 +682,16 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                     field_map.get(&field.name)
                         .map(|f| self.field_ty(span, f, substs))
                         .unwrap_or_else(|| {
-                            span_err!(tcx.sess, span, E0026,
-                                "struct `{}` does not have a field named `{}`",
-                                tcx.item_path_str(variant.did),
-                                field.name);
+                            struct_span_err!(tcx.sess, span, E0026,
+                                             "struct `{}` does not have a field named `{}`",
+                                             tcx.item_path_str(variant.did),
+                                             field.name)
+                                .span_label(span,
+                                            &format!("struct `{}` does not have field `{}`",
+                                                     tcx.item_path_str(variant.did),
+                                                     field.name))
+                                .emit();
+
                             tcx.types.err
                         })
                 }

--- a/src/test/compile-fail/E0026.rs
+++ b/src/test/compile-fail/E0026.rs
@@ -16,6 +16,8 @@ struct Thing {
 fn main() {
     let thing = Thing { x: 0, y: 0 };
     match thing {
-        Thing { x, y, z } => {} //~ ERROR E0026
+        Thing { x, y, z } => {}
+        //~^ ERROR struct `Thing` does not have a field named `z` [E0026]
+        //~| NOTE struct `Thing` does not have field `z`
     }
 }

